cmd/containerd/command/oci-hook.go

```diff
@@ -56,25 +57,6 @@ var ociHook = cli.Command{
 	},
 }
 
-type hookSpec struct {
-	Root struct {
-		Path string `json:"path"`
-	} `json:"root"`
-}
```

Comment:
Looking at this local struct, I wonder if it was specifically created for performance reasons.

The new code will unmarshal the whole spec, which is known to be resource intensive; using a shallow struct can make a big difference. See https://github.com/containerd/containerd/pull/7376 for a similar discussion / PR

Reply:
This code was written in https://github.com/containerd/containerd/pull/2418. I do not see any benchmarks there that would justify increased code complexity. Did `oci.Spec` exist back then at all?

Reply:
I'm not heavily insisting on merge of this PR. This looks like a usual "maintainability vs performance" case when the side who advocates for performance needs to prove that performance matters in this particular place so much that it is worth adding more code.

Reply:
Heh, I also went looking for the original commit, and it doesn't call it out (that said, I also know the contributor, and know they are not always spending time on calling out such things 😉)

I'm (personally) not against moving the bits to a utility for the other part; just thought I'd call out this specific one, as it's possible there was a reason

(and if we decide to keep this one, that _definitely_ warrants a comment in the code to describe "why")